### PR TITLE
Did the commands  not sure how we are going to handle all the AI stuff in the model

### PR DESCRIPTION
**Pull request by emeraldarcher**

### CODE REVIEW

**Comment by CaptBoscho** [comment]
Says it has conflicts that need to be resolved first.

**Comment by emeraldarcher** [comment]
Fixed the conflicts

File: java/src/server/facade/ServerFacade.java [hunk 1]
```diff
@@ -93,8 +102,9 @@ public void addAI(int gameID, Object aiType) throws AddAIException {
      * @throws ListAIException
      */
     @Override
-    public void listAI() throws ListAIException {
-
+    public ListAIDTO listAI(int gameId) throws ListAIException {
+        Game game = gameManager.getGameByID(gameId);
+        return new ListAIDTO(game.getAITypes());
     }
```
Comment (dannyharding10) [inline java/src/server/facade/ServerFacade.java]: Make an AIManager class that handles logic for AIs, then put `getAITypes()` in that class

File: java/src/shared/model/game/Game.java [hunk 2]
```diff
@@ -277,6 +278,15 @@ public void nextPhase() {
     //region Can do methods
     //============================================================================================
     /**
+     * Determines if an AI Player can be added to the game
+     * @return
+     */
+    @Override
+    public boolean canAddAI(){
+        return isFull();
```
Comment (dannyharding10) [inline java/src/shared/model/game/Game.java]: `return !isFull()`

File: java/src/shared/model/game/Game.java [hunk 3]
```diff
@@ -580,6 +590,20 @@ public boolean canFinishTurn(int playerIndex) {
     //region Do methods
     //=========================================================================================
     /**
+     * Adds an AI player to the game
+     *
+     * @param type
+     */
+    @Override
+    public void addAI(AIType type){
+        if(!isFull()){
```
Comment (dannyharding10) [inline java/src/shared/model/game/Game.java]: canAddAI()

File: java/src/shared/model/game/Game.java [hunk 4]
```diff
@@ -1376,6 +1400,25 @@ public CatanColor getPlayerColorByName(String player) {
     public CatanColor getPlayerColorByIndex(int index) throws PlayerExistsException {
         return this.playerManager.getPlayerColorByIndex(index);
     }
+
+    /**
+     * Get the AI types allowed
+     *
+     * @return
+     */
+    @Override
```
Comment (dannyharding10) [inline java/src/shared/model/game/Game.java]: This function should be moved to AIManager class

**Pull request merged**